### ISSUE TITLE
Why was a service provider added?

### ISSUE THREAD

**Issue by JosephSilber** [rap2hpoutre/laravel-log-viewer#16]
You don't need to register your own views namespace. Just use this [in your controller](https://github.com/rap2hpoutre/laravel-log-viewer/blob/master/src/controllers/LogViewerController.php#L17):

```php
    return View::file(__DIR__.'/../views/log.blade.php', [...]);
```

If the reason it was added is to support L4, I have to disagree. Taylor has said many times that if you need to support L4 and L5, they should live in separate branches.

Adding a service provider complicates this package for no reason. You usually don't want this in production, so I loaded this package in composer's `require-dev` array, and have this in my routes file:

```php
if (config('app.debug'))
{
	get('logs', '\Rap2hpoutre\LaravelLogViewer\LogViewerController@index');
}
```

There's no reason you should now have to conditionally load a service provider.

**Comment by rap2hpoutre** [comment]
Hi,

Thanks for your feedback. The service provider was added by this pull request (https://github.com/rap2hpoutre/laravel-log-viewer/pull/11). The reason we add this is not to add support for Laravel 4 (this log viewer supports Laravel 4 anyway). The reason was some people wanted to add their custom view (example: https://github.com/rap2hpoutre/laravel-log-viewer/issues/9), and it was not easy. So now, thanks to @AnthonyPorthouse, it's possible. About your problem, if you don't want to use this package in production environnement, you can conditionally load the service provider.

In other words, I don't want to remove the easy "customizing view" possibility, I think it's a good improvement. I understand your problem, but I think it's not critical. I did not know Taylor Otwell said we should support L4 and L5 in separate branch, but I need to understand why. I made this package compatible with L4 and L5 because I wanted to make it simple as possible for users (that was my first goal from the beginning of this project)

@JosephSilber @AnthonyPorthouse What's your opinion ?
Thanks for your feedback, hope we will find a solution.

**Comment by AnthonyPorthouse** [comment]
Conditionally loading a service provider is something you'd have to do to any packages you wanted to only use within development. For example with [barryvdh/laravel-debugbar](https://github.com/barryvdh/laravel-debugbar) you have to add a service provider so it has the correct bindings and can correctly hook itself into the system.

In this case the service provider quite literally just registers the view location against the package name so we can override them.

I do agree that for most packages the versions of Laravel should be separated away, especially if making the packager support L5 would hugely break backwards compatibility. However in this case as the package itself is very simple and hasn't even hit 1.0.0 then keeping a branch just for L4 would be overkill and perhaps even detrimental because it could massively complicate trying to add the package through composer (Is the most recent version L4 or L5?)

**Comment by xwiz** [comment]
@JosephSilber My guess is that the package would still grow to the point where there will be need for a ServiceProvider. I wanted a configurable access key to access the log view for example and ended up having to register commands and include package routes. This way it's considerably easier to dispose of the package and stuff associated with it when no longer needed as well as get it running faster once installed.
